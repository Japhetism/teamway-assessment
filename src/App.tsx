import React, { useEffect, useState } from "react";
import {
  BrowserRouter,
  Outlet,
  Route,
  Routes
} from "react-router-dom";
import Navigation from "./components/navigation";
import Background from "./components/background";
import Landing from "./features/landing/ui";
import Test from "./features/test/ui";
import Result from "./features/result/ui";
import NotFound from "./features/notfound/ui";
import './App.css';

export function App(): JSX.Element {
  useEffect(() => {
    switch (localStorage.theme) {
      case "dark":
        document.documentElement.classList.add("dark");
        break;
      case "light":
        document.documentElement.classList.remove("dark");
        break;
      default:
        if (window.matchMedia("(prefers-color-scheme: dark)").matches)
          document.documentElement.classList.add("dark");
        else document.documentElement.classList.remove("dark");
        break;
    }
  }, []);

  return (
    <Background className="h-auto w-auto">
      <Navigation>
        <BrowserRouter>
          <Routes>
            <Route index element={<Landing />} />
            <Route path="/personality/take-test" element={<Test />} />
            <Route path="/personality/result" element={<Result />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </BrowserRouter>
      </Navigation>
    </Background>
  );
}

export default App;
