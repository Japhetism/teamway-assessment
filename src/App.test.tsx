import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import App from "./App";

const renderer = createRenderer();

test("should render App and match the snapshot", () => {
  renderer.render(<App />);
  const renderedOutput = renderer.getRenderOutput();

  expect(renderedOutput).toMatchSnapshot();
});
