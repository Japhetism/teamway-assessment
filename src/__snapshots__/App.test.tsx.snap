// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App and match the snapshot 1`] = `
<Background
  className="h-auto w-auto"
>
  <Navigation>
    <BrowserRouter>
      <Routes>
        <Route
          element={<Landing />}
          index={true}
        />
        <Route
          element={<Test />}
          path="/personality/take-test"
        />
        <Route
          element={<Result />}
          path="/personality/result"
        />
        <Route
          element={<NotFound />}
          path="*"
        />
      </Routes>
    </BrowserRouter>
  </Navigation>
</Background>
`;
